midterm-2023-11-07/code/5048-dijana-damcevska/scripts.ts

```diff
@@ -0,0 +1,225 @@
+
+    const genreElement = document.getElementById("genre-filter") as HTMLSelectElement;
+    const genre = genreElement.value;
+
+    const yearElement = document.getElementById("year-filter") as HTMLSelectElement;
+    const year = yearElement.value;
+
+    const directorElement = document.getElementById("director-filter") as HTMLSelectElement;
+    const director = directorElement.value.toLowerCase().trim();
+
+    let filteredMovies = movies;
+    if (genre !== "all") {
+        filteredMovies = filteredMovies.filter(movie => movie.genre.join().toLowerCase().includes(genre.toLowerCase()));
+    }
+
+    if (title !== "") {
+        filteredMovies = filteredMovies.filter(movie => movie.title.toLowerCase().includes(title));
+    }
+
+    if (year !== "") {
+        filteredMovies = filteredMovies.filter(movie => movie.year.toString() === year);
+    }
+
+    if (director !== "") {
+        filteredMovies = filteredMovies.filter(movie => movie.director.toLowerCase().includes(director));
+    }
+
+    displayMovies(filteredMovies);
+}
+
+
+const loadData = async () => {
+    const url = 'https://raw.githubusercontent.com/sweko/uacs-internet-programming-exams/main/midterm-2023-11-07/data/movies.json';
+    const response = await fetch(url);
+    if (!response.ok) {
+        throw new Error("Response Not OK")
+    }
+
+    const data = await response.json();
+    return data;
+}
+
+const displayMovies = (movies: Movie[]) => {
+
+    const container = document.getElementById('movie-container')!;
+    container.innerHTML = "";
+    for (const movie of movies) {
+        const movieRow = generateMovieRow(movie);
+        container.appendChild(movieRow);
+    }
+}
+
+
+const generateMovieRow = (movie: Movie) => {
+    const row = document.createElement('div');
+    row.classList.add("movie-row")
+
+    const idCell = document.createElement('div');
+    idCell.classList.add('movie-data', 'movie-id');
+    idCell.innerHTML = movie.id.toString();
+    row.appendChild(idCell);
+
+    const titleCell = document.createElement('div');
+    titleCell.classList.add('movie-data', 'movie-title');
+    titleCell.innerHTML = movie.title;
+    row.appendChild(titleCell);
+
+    const yearCell = document.createElement('div');
+    yearCell.classList.add('movie-data', 'movie-year');
+    yearCell.innerHTML = movie.year.toString();
+    row.appendChild(yearCell);
+
+    const directorCell = document.createElement('div');
+    directorCell.classList.add('movie-data', 'movie-director');
+    directorCell.innerHTML = movie.director;
+    row.appendChild(directorCell);
+
+    const genreCell = document.createElement('div');
+    genreCell.classList.add('movie-data', 'movie-genre');
+    genreCell.innerHTML = movie.genre.join(' / ');
+    row.appendChild(genreCell);
+
+    const plotCell = document.createElement('div');
+    plotCell.classList.add('movie-data', 'movie-plot');
+
+    let cutPlot = movie.plot.substring(0, 50);
```

Comment:
This needs to be extracted to a separate method, so that it can be tested better, and does not mess up our nice cell generation code. +8